doc/sphinx/proofs/writing-proofs/inductive.rst

```diff
@@ -280,49 +285,48 @@ analysis on inductive or co-inductive objects (see :ref:`inductive-definitions`)
 .. exn:: Not an inductive product.
    :undocumented:
 
-.. exn:: Unable to find an instance for the variables @ident ... @ident.
+.. exn:: Unable to find an instance for the variables @ident … @ident.
 
    Use in this case the variant :tacn:`elim … with` below.
 
-.. tacv:: induction @term as @or_and_intropattern_loc
+.. tacv:: induction @term @as_or_and_ipat
 
-   This behaves as :tacn:`induction` but uses the names in
-   :n:`@or_and_intropattern_loc` to name the variables introduced in the
-   context. The :n:`@or_and_intropattern_loc` must typically be of the form
-   :n:`[ p` :sub:`11` :n:`... p` :sub:`1n` :n:`| ... | p`:sub:`m1` :n:`... p`:sub:`mn` :n:`]`
+   Behaves like :tacn:`induction` but uses the names in
+   :n:`@as_or_and_ipat` to name the variables introduced in the
+   context. The :n:`@as_or_and_ipat` must typically be of the form
```

Comment:
@Zimmi48, any comment?  @JasonGross?

Reply:
Or a simpler example:  Here I don't see how you would know that "m" and "IHX" are associated with `nat` and the hypothesis and not vice-versa, let alone for the 7 variable example above.

```
Goal forall n:nat, n + 0 = n.
induction n as [|m IHX]. 2: {
(* 1 subgoal
m : nat
IHX : m + 0 = m
______________________________________(1/1)
S m + 0 = S m *)

Check nat_ind.
(* nat_ind
     : forall P : nat -> Prop,
       P 0 ->
       (forall n : nat, P n -> P (S n)) ->
       forall n : nat, P n *)
```

Reply:
One thing that is always easy to guess in the `as` clause structure for `destruct`, `induction` and `inversion` is the number of cases separated with `|` (the same as the number of constructors).
For `destruct`, it's also relatively easy to guess the number and the order of the variables because it corresponds to the number and order of arguments of each constructor.
For `induction`, this is a bit more complex, but you can still guess by looking at the form of the induction principle. E.g.:

```coq
nat_ind
     : forall P : nat -> Prop, P 0 -> (forall n : nat, P n -> P (S n)) -> forall n : nat, P n
```

There are two premises which correspond to the two constructors. The first premise has no (dependent or non-dependent) premise of its own, while the second has `n`, followed by `P n`, so you need to provide a name for the variable and a name for the (induction) hypothesis.

On the earlier more complex example:

```coq
clos_refl_trans_ind
     : forall (A : Type) (R : relation A) (P : A -> A -> Prop),
       (forall x y : A, R x y -> P x y) ->
       (forall x : A, P x x) ->
       (forall x y z : A, R* x y -> P x y -> R* y z -> P y z -> P x z) -> forall x a : A, R* x a -> P x a
```

The premise which corresponds to the third constructor is:

```coq
forall x y z : A, R* x y -> P x y -> R* y z -> P y z -> P x z
```

You have variables (dependent premises) followed by four hypotheses (non-dependent premises).